tests/cluster/etl/test_transformer.py

```diff
@@ -14,19 +14,70 @@ def setUpClass(cls):
         cls.transformer = TestTransformer()
         cls.df = create_dataframe()
 
-    def test_process_returns_dataframe(self):
+    def test_process(self):
+        # Process is called with single input
+        # Result contain single df
         result = self.transformer.etl({"df": self.df})
+        self.assertEqual(1, len(result))
         self.assertEqual({"TestTransformer"}, set(result.keys()))
-        self.assertIs(list(result.values())[0], self.df)
+        self.assertIs(self.df, result["TestTransformer"])
 
     def test_process_many(self):
+        # Process_many is called with multiple inputs
+        # Result contain union of input dfs og length 6
         result = self.transformer.etl({"df1": self.df, "df2": self.df})
+        self.assertEqual(1, len(result))
         self.assertEqual({"TestTransformer"}, set(result.keys()))
-        self.assertEqual(6, list(result.values())[0].count())
+        self.assertEqual(6, result["TestTransformer"].count())
 
     def test_dataset_key(self):
-        result = TestTransformer("my_key").etl({"df": self.df})
+        # Set dataset_key to control name of the output
+        # Result contain the key 'my_key'
+        result = TestTransformer(dataset_key="my_key").etl({"df": self.df})
+        self.assertEqual(1, len(result))
         self.assertEqual({"my_key"}, set(result.keys()))
+        self.assertIs(self.df, result["my_key"])
+
+    def test_dataset_output_key(self):
+        # Set dataset_output_key to control name of the output
+        # Result contain the key 'my_key'
+        result = TestTransformer(dataset_output_key="my_key").etl({"df": self.df})
+        self.assertEqual(1, len(result))
+        self.assertEqual({"my_key"}, set(result.keys()))
+        self.assertIs(self.df, result["my_key"])
+
+    def test_single_dataset_input_key(self):
+        # Set dataset_input_keys to control what dfs in dataset to handle
+        # Process is called with single input
+        # Result contain two keys due to 'df2' not being consumed
+        # when not in dataset_input_keys
+        result = TestTransformer(dataset_input_keys="df1").etl(
+            {"df1": self.df, "df2": self.df}
+        )
+        self.assertEqual(2, len(result))
+        self.assertEqual({"TestTransformer", "df2"}, set(result.keys()))
+        self.assertIs(self.df, result["TestTransformer"])
+
+    def test_multiple_dataset_input_keys(self):
+        # Set dataset_input_keys to control what dfs in dataset to handle
+        # Process_many is called with multiple inputs
+        # Result contain two keys due to 'df3' not being consumed
```

Comment:
If df3 is not consumed, wouldnt it then contain THREE keys?

Reply:
'df3' is not consumed, but 'df1' and 'df2' are merged into 'TestTransformer' therefore 2 keys.